spring-cloud-gcp-data-firestore/src/test/java/com/google/cloud/spring/data/firestore/repository/query/FirestoreRepositoryTests.java

```diff
@@ -70,7 +71,7 @@ public void testSortQuery() {
 				Order.newBuilder()
 						.setDirection(Direction.ASCENDING)
 						.setField(
-								FieldReference.newBuilder().setFieldPath("name"))
+								FieldReference.newBuilder().setFieldPath(FieldPath.documentId().toString()))
```

Comment:
This is an interesting change. The former line used "name"; the current uses "__name__", so at first glance, it looks like a behavior change. But! Notice, this is a unit test (well, it's a unit test for a fairly large "unit" extending from "spring data parses repository method name" to "firestore template execute called"). So while this test "used to work", this same test would fail in an integration test setup. 

So this got me to wonder... Did our sort by ID property works at all? And when I wonder whether something works, I write an integration test :

_FirestoreRepositoryIntegrationTests_
```
	@Test
	public void testOrderByDocumentId() {
		User alice = new User("Alice", 99);
		User bob = new User("Bob", 99);
		User zelda = new User("Zelda", 99);
		this.userRepository.save(alice)
				.then(this.userRepository.save(bob))
				.then(this.userRepository.save(zelda))
				.block();

		StepVerifier.create(
				this.userRepository.findByAge(99, Sort.by(Order.desc("name"))).map(User::getName)
		).expectNext("Zelda", "Bob", "Alice")
				.verifyComplete();
	}
```

So, get this: the test fails on `main` branch, and passes on your pull request. Congratulations on fixing a bug we never knew we had!